### PR TITLE
fix input shape of op tests

### PR DESCRIPTION
**Pull request by zhupengyang**
- fix input shape of op tests for elementwise_max, elementwise_sub, gather, hierarchical_sigmoid, pad2d, reduce_sum, reshape2, reverse, squeeze2, squeeze, sum, transpose, unsqueeze, unsqueeze2
- fix https://github.com/PaddlePaddle/Paddle/pull/21391/files#r356043791

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/test_cast_op.py [hunk 1]
```diff
@@ -24,7 +24,7 @@
 
 class TestCastOp1(op_test.OpTest):
     def setUp(self):
-        ipt = np.random.random(size=[10, 10])
+        ipt = np.random.random(size=[10, 12])
```
Comment (luotao1) [inline python/paddle/fluid/tests/unittests/test_cast_op.py]: 规则是否改成大于等于100即可？

File: python/paddle/fluid/tests/unittests/test_elementwise_max_op.py [hunk 2]
```diff
@@ -58,17 +58,17 @@ def setUp(self):
 class TestElementwiseMaxOp_Vector(TestElementwiseOp):
     def setUp(self):
         self.op_type = "elementwise_max"
-        x = np.random.random((32, )).astype("float32")
-        sgn = np.random.choice([-1, 1], (32, )).astype("float32")
-        y = x + sgn * np.random.uniform(0.1, 1, (32, )).astype("float32")
+        x = np.random.random((110, )).astype("float32")
```
Comment (luotao1) [inline python/paddle/fluid/tests/unittests/test_elementwise_max_op.py]: 110是否是个比较不通用的维度数据？

File: python/paddle/fluid/tests/unittests/test_gather_op.py [hunk 3]
```diff
@@ -51,7 +51,7 @@ def config(self):
         """
         For one dimension input
         """
-        self.x_shape = (10)
+        self.x_shape = (10, 20)
```
Comment (luotao1) [inline python/paddle/fluid/tests/unittests/test_gather_op.py]: 为了CI效率考虑，尽量不要太大，比如这里设置`(10,10)`即可。下同。

File: python/paddle/fluid/tests/unittests/test_cast_op.py (hunk 1 above)
Comment (zhupengyang) [reply python/paddle/fluid/tests/unittests/test_cast_op.py]: Done

File: python/paddle/fluid/tests/unittests/test_elementwise_max_op.py (hunk 2 above)
Comment (zhupengyang) [reply python/paddle/fluid/tests/unittests/test_elementwise_max_op.py]: 改为100。Done

File: python/paddle/fluid/tests/unittests/test_gather_op.py (hunk 3 above)
Comment (zhupengyang) [reply python/paddle/fluid/tests/unittests/test_gather_op.py]: Done

**Pull request merged**